integreat_cms/cms/utils/external_calendar_utils.py

```diff
@@ -73,15 +74,25 @@ def from_ical_event(
             content[:32],
         )
 
+        is_all_day = not (
+            isinstance(start, datetime.datetime) and isinstance(end, datetime.datetime)
+        )
+        if is_all_day:
+            start_date, start_time = start, None
+            end_date, end_time = end - datetime.timedelta(days=1), None
+        else:
+            start_date, start_time = start.date(), start.time()
+            end_date, end_time = end.date(), end.time()
```

Comment:
Do you think we can put this into it's own method to make the method a bit smaller? :)

Reply:
I originally wanted to say "yes, sure!", but - that function would either have to take `start` and `end` as input and return `start_date`, `start_time`, `end_date`, `end_time`, `is_all_day` as a giant 5-tuple, or be at least three different functions.

Not sure if that is wirth it? 😅  If you insist I'll implement it, I'm just a bit ambivalent here. Small functions are nice, but I am not sure it would help with readability.

Reply:
Oops, I didn't realize I was blocking this PR :/ To be honest, at least for now, I don't have a clear opinion or idea. To me this is another function that keeps growing and eventually will be hard to maintain. However I also see your point and you're absolutely right. So I don't know :shrug:

Reply:
> To me this is another function that keeps growing and eventually will be hard to maintain.

That's absolutely valid. The only thing I can really give as a counterargument is that it's not a short function, but it's also not a complex function - no loops, and this is the only if/else branch. *So far* it's OK to maintain, I think :D

But I'll leave it up to you.  If the above is not convincing, LMK, and I'll use this PR to try and refactor the function.

Reply:
> So far it's OK to maintain, I think :D
Yeah, on second thought I agree. I think we can merge this PR :)